salvo/src/lib/test_source_tree.py

```diff
@@ -9,83 +9,240 @@
 from src.lib import (cmd_exec, source_tree)
 import api.source_pb2 as proto_source
 
+logging.basicConfig(level=logging.DEBUG)
 log = logging.getLogger(__name__)
 
 def test_source_tree_object():
   """Verify that we throw an exception if not all required data is present."""
-  pass
+  st = source_tree.SourceTree(proto_source.SourceRepository())
+
+  with pytest.raises(source_tree.SourceTreeError) as pull_exception:
+    st.get_head_hash()
+
+  assert "No origin is defined or can be" in str(pull_exception.value)
+
 
 def test_git_with_origin():
   """Verify that at a minimum, we can work with a remote origin url
      specified.
   """
-  pass
+  source_repository = proto_source.SourceRepository(
+      source_url='somewhere_in_github'
+  )
+  st = source_tree.SourceTree(source_repository)
+
+  assert st.get_origin()
+
 
 def test_source_tree_with_local_workdir():
   """Verify that we can work with a source location on disk."""
-  pass
+  source_repository = proto_source.SourceRepository(
+      source_path='/some_source_path'
+  )
+  st = source_tree.SourceTree(source_repository)
+
+  remote_string = 'origin  git@github.com:username/reponame.git (fetch)'
+
+  with mock.patch('src.lib.cmd_exec.run_command',
+                  mock.MagicMock(return_value=remote_string)) as magic_mock:
+    origin = st.get_origin()
+    assert origin == "git@github.com:username/reponame.git"
+
+    cmd_params = cmd_exec.CommandParameters(cwd='/some_source_path')
+    magic_mock.assert_called_once_with("git remote -v", cmd_params)
 
 def test_get_origin_ssh():
   """Verify that we can determine the origin for a local repository.
 
   In this instance the repo was cloned via ssh.
   """
-  pass
+  remote_string = 'origin  git@github.com:username/reponame.git (fetch)'
+  gitcmd = "git remote -v"
```

Comment:
(nit) The name should be `git_cmd` since these are two words.

Applies to multiple locations.

Reply:
Updated